shared/README.md

```diff
@@ -7,25 +7,57 @@
 - Service Principal
 - [Terraform](https://www.terraform.io/downloads.html)
 
+Fill the values required below in backend.tfvars to initialize Azure backend for Terraform, using the information [below](###Configure-Terraform-to-Store-State-Data-in-Azure)
+
+- Storage Account Name
+- Storage Account Access Key
+- Storage Account Container Name
+- Key Name to store for Terraform State for Test Environment
+
+
 ## Resources
 
-The following respources will be deployed
+The following resources will be deployed:
 - Azure Resource Group
 - Azure KeyVault 
 
 ## Deployment
 
-1. Authenticate using your Azure Principal or an Azure account with privileges to deploy resource groups.
+### Authenticate using your Azure Principal or an Azure account with privileges to deploy resource groups.
 
 ``` bash
 $ az login
 ```
 
-2. Execute the following commands:
+### Configure Terraform to Store State Data in Azure
+
+Terraform records the information about what is created in a [Terraform state file](https://www.terraform.io/docs/state/) after it finishes applying.  By default, Terraform will create a file named `terraform.tfstate` in the directory where Terraform is applied.  Terraform needs this information so that it can be loaded when we need to know the state of the cluster for future modifications.
+
+In production scenarios, storing the state file on a local file system is not desired because typically you want to share the state between operators of the system.  Instead, we configure Terraform to store state remotely, and in Cobalt we use Azure Blob Store for this storage.  This is defined using a `backend` block.  The basic block looks like:
+
+```bash
+terraform {
+   backend “azure” {
+   }
+}
+```
+
+In order to setup an Azure backend, one needs an Azure Storage account.  If one must be provisioned, use [this](https://docs.microsoft.com/en-us/cli/azure/storage/account?view=azure-cli-latest#az-storage-account-create) link to provision, where `name` is the name of the storage account to store the Terraform state, `location` is the Azure region the storage account should be created in, and `resource group` is the name of the resource group to create the storage account in.  
+
+If there is already a pre-existing storage account, then retrieve the equivalent information for the existing account.
+
+Once the storage account details are known, we need to fetch the storage account key so we can configure Terraform with it:
+
+```bash
+>  az storage account keys list --account-name <storage account name>
+```
+
```

Comment:
The user also needs to have the storage account container created or identified at this step.  Please add instructions/comments for this.